### PR TITLE
Storage for appointments

### LINKED ISSUE #74: Storage: Appointment
Save appointment model as a json file

### PR DESCRIPTION
**Pull request by JinHao-L**
Implemented storage for appointments. (Only add appointments that have a date after or equal to today's date)
Implemented check to ensure consistency between appointment data and patient data

Others: 
* Add sample appointment data for new user
* Add tests for appointment storage
* Update display message of view all
* Simplify Date and Time creation for tests

closes #74

### CODE REVIEW

File: src/main/java/seedu/address/commons/util/DateTimeUtil.java [hunk 1]
```diff
@@ -13,23 +13,35 @@
...
+     * @return true if given date is before today's date, false otherwise.
      */
-    public static boolean isExpired(LocalDate date, LocalTime time) {
-        return isExpired(date, time, LocalDateTime.now());
+    public static boolean isExpiredByDay(LocalDate date) {
+        return isExpired(date, LocalTime.MIDNIGHT, LocalDate.now().atStartOfDay());
     }
 
     /**
      * Check if the given date and time is expired.
```
Comment (WadeForward) [inline src/main/java/seedu/address/commons/util/DateTimeUtil.java]: should be `checks`?

File: src/main/java/seedu/address/commons/util/DateTimeUtil.java [hunk 2]
```diff
@@ -13,23 +13,35 @@
...
-     * @return true if given date and time is in the future, false otherwise.
+     * @return true if given date and time is not expired, false otherwise.
+     */
+    public static boolean isExpired(LocalDate date, LocalTime time) {
+        return isExpired(date, time, LocalDateTime.now());
+    }
+
+
+    /**
+     * Similar to {@link DateTimeUtil#isExpired(LocalDate, LocalTime)}
```
Comment (WadeForward) [inline src/main/java/seedu/address/commons/util/DateTimeUtil.java]: is period needed here?

File: src/main/java/seedu/address/logic/commands/ViewCommand.java [hunk 3]
```diff
@@ -59,6 +58,6 @@ public CommandResult execute(Model model) throws CommandException {
     public boolean equals(Object other) {
         return other == this
                 || (other instanceof ViewCommand
-                    && predicate.equals(((ViewCommand) other).predicate));
+                && predicate.equals(((ViewCommand) other).predicate));
```
Comment (WadeForward) [inline src/main/java/seedu/address/logic/commands/ViewCommand.java]: Thanks for the correction 👍

File: src/main/java/seedu/address/model/ModelManager.java [hunk 4]
```diff
@@ -207,4 +208,24 @@ public boolean equals(Object obj) {
                 && filteredAppointments.equals(other.filteredAppointments);
     }
 
+    //=========== Model Validation =============================================================
+
+    /**
+     * Check if {@code ReadOnlyAppointmentBook} is consistent with {@code ReadOnlyAddressBook} data.
+     *
+     * @param addressBook     the patients data
```
Comment (WadeForward) [inline src/main/java/seedu/address/model/ModelManager.java]: The space between `addressBook` and `the ...` is too much?

File: src/main/java/seedu/address/model/appointment/Date.java [hunk 5]
```diff
@@ -16,6 +16,17 @@
 
     private final LocalDate value;
 
+    /**
+     * Create a {@code Date} that represents the date of an appointment.
```
Comment (WadeForward) [inline src/main/java/seedu/address/model/appointment/Date.java]: Should be `Creates`?

**Review by WadeForward — COMMENTED** [review]
Great work! 👍 
Just some very minor changes need to be made.

File: src/main/java/seedu/address/commons/util/DateTimeUtil.java (hunk 1 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/commons/util/DateTimeUtil.java]: Ohh yeah. I have updated this on my side. Thanks for spotting it 👍

File: src/main/java/seedu/address/commons/util/DateTimeUtil.java (hunk 2 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/commons/util/DateTimeUtil.java]: Agreed 👍

File: src/main/java/seedu/address/logic/commands/ViewCommand.java (hunk 3 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/logic/commands/ViewCommand.java]: I think this is auto-correct on my part! It should have a tab right?

File: src/main/java/seedu/address/model/ModelManager.java (hunk 4 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/model/ModelManager.java]: I have edited this on my side. 👍

File: src/main/java/seedu/address/model/appointment/Date.java (hunk 5 above)
Comment (JinHao-L) [reply src/main/java/seedu/address/model/appointment/Date.java]: Edited! Thanks for spotting it! 👍

**Review by WadeForward — APPROVED** [review]
LGTM! 👍

**Review by xz0127 — APPROVED** [review]
Amazing work. LGTM! 👍

**Pull request merged**